torch/ao/quantization/fake_quantize.py

```diff
@@ -147,7 +150,7 @@ class FakeQuantize(FakeQuantizeBase):
     scale: torch.Tensor
     zero_point: torch.Tensor
 
-    def __init__(self, observer=MovingAverageMinMaxObserver, quant_min=None, quant_max=None, **observer_kwargs):
+    def __init__(self, observer=MovingAverageMinMaxObserver, quant_min=None, quant_max=None, is_dynamic=False, **observer_kwargs):
```

Comment:
I also dont understand what it means to have is_dynamic for moving average min/max observer. Doesnt feel right

Reply:
It feel like only MinMaxObserver/HistogramObserver can have is_dynamic. Anything that requires larger window of observeration cannot be called dynamically quantized? So thats my question.

Reply:
My question is this: For dynamic quant, can we not force that the observer must be Histogram or MinMax, not moving average. Not sure if thats what you answered above

Reply:
right now we force it must be MovingAverageMinMaxObserver with averaging_constant=1, this is the observer+configuration that mimics the computation of dynamic quant exactly in QAT and after convert (which is choose_qparams on a single input + q + dq). normal minmax observer or histogram observer does not have this behavior so they do not have corresponding dynamic quant simulation/conversion.

Reply:
minmax/histogram works for static QAT.

this is just talking about dynamic quantization (QAT/PTQ), so currently in convert, the pattern we produce for dynamic quantization is `choose_qparams -> q -> dq` and the computation is equivalent to AveragingMinMaxObserver with averaging_constant = 1.. 

all other computations does not have a corresponding dynamic quant pattern, e.g. histogram observer/minmax observer or averaging minmax observer with different averaging_constant all requires observing multiple inputs and maintaining some states, this is different from dynamic quantization which does not have any state, that's why MovingAverageMinMaxObserver with averaging_constant=1 (meaning no state) is the only thing that corresponds to the current implementation of dynamic quantization.